crypto/evp/digest.c

```diff
@@ -70,6 +70,42 @@ int EVP_MD_CTX_reset(EVP_MD_CTX *ctx)
     return 1;
 }
 
+EVP_MD_CTX *evp_md_ctx_new_ex(EVP_PKEY *pkey, const ASN1_OCTET_STRING *id,
+                              OPENSSL_CTX *libctx, const char *propq)
+{
+    EVP_MD_CTX *ctx;
+    EVP_PKEY_CTX *pctx = NULL;
+
+    if ((ctx = EVP_MD_CTX_new()) == NULL
+        || (pctx = EVP_PKEY_CTX_new_from_pkey(libctx, pkey, propq)) == NULL) {
+        X509err(0, ERR_R_MALLOC_FAILURE);
+        goto err;
+    }
+
+#ifndef OPENSSL_NO_SM2
+# ifndef FIPS_MODULE /* TODO(3.0): EVP_PKEY_CTX_ctrl() not yet supported in FIPS module */
```

Comment:
`evp_md_ctx_new_ex` can't be made static because it's also used by a_verify.c.
But good thought to move it to `crypto/asn/` - it now lives in a_verify.c,
and its declaration is now in include/crypto/asn1.h